Elasticache-cluster terraform module update: support for Elasticache-cluster disabled ref. SAAS-11653 (after rebase)

rameshganne opened the pull request:
The latest Elasticache-cluster terraform module & other files can create AWS Elasticache-cluster disable mode in Staging environment. Present, it can supports for cluster-disabled mode only.

Here, we added "create option" in terraform [environments/staging/terraform.yml] environments files and creates infra based on create key value i.e. yes|no.
Updated files other than terraform module:

environments/staging/terraform.yml
environments/india/terraform.yml
environments/production/terraform.yml
file - src/commcare_cloud/commands/terraform/templates/commcarehq.tf.j2 : updated file to create commcarehq.tf with latest attributes
file - src/commcare_cloud/environment/schemas/terraform.py: updated file to get the latest values from environments-terraform files.[ example: environments/staging/terraform.yml ]

Similar to PR: https://github.com/dimagi/commcare-cloud/pull/4415  https://github.com/dimagi/commcare-cloud/pull/4417

dannyroberts commented on environments/staging/terraform.yml:
```diff
@@ -249,18 +249,20 @@ elasticache:
 
 r53_private_zone:
+  create: yes
   domain_name:  "stg.dimagi.local"
```
I'm curious what this is used for and whether we can give it a better name. Since the networks between environments are totally isolated, if this is the endpoint other machines will use to access redis, perhaps we can just call it "redis.local"? Or if it's good to be more explicit, we can call it "redis.staging.local" here (and "redis.production.local", "redis.india.local" on the other environments?)

Something worth getting right before we apply on production, but no need to block this PR

dannyroberts replied:
we agreed to `*.staging.commcare.local` so in this case `redis.staging.commcare.local`

rameshganne replied:
I have updated domain name [ staging.commcare.local ] and added "create_record option" in terraform [environments/staging/terraform.yml] environment file and creates Route-53 record based on create_record key value i.e. yes|no.

rameshganne commented:
I have updated domain name [ staging.commcare.local ] and added "create_record option" in terraform [environments/staging/terraform.yml] environment file and creates Route-53 record based on create_record key value i.e. yes|no.
